src/main/java/pl/edu/mimuw/chess/pieces/MoveUtil.java

```diff
@@ -0,0 +1,76 @@
+package pl.edu.mimuw.chess.pieces;
+
+import pl.edu.mimuw.chess.Board.Board;
+import pl.edu.mimuw.chess.Board.Field;
+
+import java.util.List;
+
+public class MoveUtil {
+
+  public static boolean addWhenCorrect(Piece piece, Board board, Field moveTo, List<Field> moves){
+    if(moveTo.isCorrect()) {
+      var moveDestination = board.getPiece(moveTo);
+      if (moveDestination == null || !moveDestination.getTeam().equals(piece.getTeam())) {
+        moves.add(moveTo);
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public static void moveDiagonal(int range, Piece piece, Board board, List<Field> moves){
+    int i;
+
+    i=1;
+    while (addWhenCorrect(piece, board, piece.getField().se(i), moves)) {
```

Comment:
ta petla z while powinna zostać wydzielona do funkcji pomocniczej, gdzie jako jeden z argumentów brałaby argument funkcyjny w postaci funkcji `Field::se` - zdecydowanie polecam spróbować to zrobić, jako wskazóskę podpowiem że typ parametru to byłoby chyba `Function<Field, Field>`